Build on latest nixpkgs-unstable fails

<!-- Please describe the issue here at the top, then fill in the system information below. -->

### System Information

<!-- Paste the output of `niri -V`, e.g. niri 0.1.0-beta.1 (v0.1.0-beta.1) -->
* niri version: e251ca7 (current HEAD)

<!-- Write your distribution, e.g. Fedora 40 Silverblue -->
* Distro: NixOS 6.13.5

After updating my machine configuration niris build fails with the following error:

```
       >   = note: LC_ALL="C" PATH="/nix/store/464j7y43imzxc94955xgysxxcnadaah0-rustc-1.85.0/lib/rustlib/x86_64-unknown-linux-gnu/bin:/nix/store/wvcl683adl5vsc2873rdcs255zq3j7yk-cargo-1.85.0/bin:/nix/store/q88m6mq2pbd8nq7dr0p065334nsglqhr-cargo-auditable-0.6.5/bin:/nix/store/8cczxvv6hpngc9mxal0wx0lzny4vjp4n-pkg-config-wrapper-0.29.2/bin:/nix/store/2b3ahc691h5syn9qp18fkps6hp1n0rdl-auditable-cargo-1.85.0/bin:/nix/store/fsyj2gv51gcwcs8gx44xix044hb716a5-rustc-wrapper-1.85.0/bin:/nix/store/wvcl683adl5vsc2873rdcs255zq3j7yk-cargo-1.85.0/bin:/nix/store/qp4a5bxdck7c0w7l2mf47phhfk2nmscs-patchelf-0.15.0/bin:/nix/store/gj9lra51hwhxnhz05jqk5lh03wipamv0-gcc-wrapper-14-20241116/bin:/nix/store/aw0qxjd1phf16qhlwpdb4x87yymfv9rp-gcc-14-20241116/bin:/nix/store/cgp5cbv6b3djw35zkl4s77snh0navr3w-glibc-2.40-66-bin/bin:/nix/store/vz67zaqyfln2jcqwar682kdlrxfx9vq1-coreutils-9.6/bin:/nix/store/kinjsav83qgiiy8mwgypnnc9sn8m8r9y-binutils-wrapper-2.43.1/bin:/nix/store/dh0m67pv8snfcx1ic18jzp0x4cb10zhc-binutils-2.43.1/bin:/nix/store/vz67zaqyfln2jcqwar682kdlrxfx9vq1-coreutils-9.6/bin:/nix/store/cfd3hysxk7b4m584s539hzm2bc013lsg-findutils-4.10.0/bin:/nix/store/kdrp6y7k352jb9ysjprzvmrlmiblz0jq-diffutils-3.10/bin:/nix/store/l97jblhqk9ga2dl01ysd3wf51mwqcd6s-gnused-4.9/bin:/nix/store/h3xhfhg0vv3k3hvkcwkmdg99r4zm9mcl-gnugrep-3.11/bin:/nix/store/kz9cfpxf08fdsgxgh6zp1k1y5npsnjpr-gawk-5.3.1/bin:/nix/store/j9gjzsbag34n3ic3yg4ra0p5yl8wcbhg-gnutar-1.35/bin:/nix/store/g9h265xkvsma74irj4389djs39k96d72-gzip-1.13/bin:/nix/store/p406mc3r448lfrl0qgqm4cg8qax7jw5v-bzip2-1.0.8-bin/bin:/nix/store/zkman1hvq58ica17s06r9hq62k300c1v-gnumake-4.4.1/bin:/nix/store/ki4if6b0w5bqv8dc5lrjp8xm7wjy9dlf-bash-5.2p37/bin:/nix/store/gkw3fwmfvn06gz7hh6vwdskh3s1lkvcv-patch-2.7.6/bin:/nix/store/7w6d3dwm5zs6vn37wmsbi0szvx0w98gy-xz-5.6.4-bin/bin:/nix/store/i4fn3frirjgrfsmwkbr4srfpry18x5dz-file-5.46/bin" VSLANG="1033" "/nix/store/gj9lra51hwhxnhz05jqk5lh03wipamv0-gcc-wrapper-14-20241116/bin/cc" "-m64" "/build/rustcLxMzHl/symbols.o" "<1218 object files omitted>" "-Wl,--as-needed" "-Wl,-Bstatic" "/build/rustcLxMzHl/{liblibspa-f3a1e5f33a23c982.rlib,liblibspa_sys-3b16dab4049d9995.rlib}" "/nix/store/464j7y43imzxc94955xgysxxcnadaah0-rustc-1.85.0/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libcompiler_builtins-b15b4ec8580e3dd7.rlib}" "-Wl,-Bdynamic" "-lpipewire-0.3" "-lpipewire-0.3" "-lpangocairo-1.0" "-lpango-1.0" "-lgobject-2.0" "-lglib-2.0" "-lharfbuzz" "-lcairo" "-lpango-1.0" "-lgobject-2.0" "-lglib-2.0" "-lharfbuzz" "-lgio-2.0" "-lgobject-2.0" "-lglib-2.0" "-lgobject-2.0" "-lglib-2.0" "-lcairo-gobject" "-lcairo" "-lgobject-2.0" "-lglib-2.0" "-lgobject-2.0" "-lglib-2.0" "-ldisplay-info" "-lseat" "-ludev" "-linput" "-ldl" "-lpixman-1" "-lxkbcommon" "-ldl" "-lgbm" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/nix/store/48jgb2ihb9ccgqmwcdgwkg8vbrwcn169-systemd-minimal-libs-257.3/lib" "-L" "/nix/store/sy0n827w0ccnl4izd50qbijmnc9np852-libdisplay-info-0.2.0/lib" "-L" "/nix/store/22svsj3ivbibhksnd7lwawdnk7wgbz9a-seatd-0.9.1/lib" "-L" "/nix/store/7ly9p55j21sr0rl93lb6vpi1k1sbb3x3-glib-2.82.4/lib" "-L" "/nix/store/7ly9p55j21sr0rl93lb6vpi1k1sbb3x3-glib-2.82.4/lib" "-L" "/nix/store/7ly9p55j21sr0rl93lb6vpi1k1sbb3x3-glib-2.82.4/lib" "-L" "/nix/store/7ly9p55j21sr0rl93lb6vpi1k1sbb3x3-glib-2.82.4/lib" "-L" "/nix/store/7ly9p55j21sr0rl93lb6vpi1k1sbb3x3-glib-2.82.4/lib" "-L" "/nix/store/3fnl15840l1pjj7k6yypwv3hxdsp5wam-pango-1.56.1/lib" "-L" "/nix/store/zr9wda9pmmzrkv8hisck3vplcpds7k7b-harfbuzz-10.2.0/lib" "-L" "/nix/store/0713644mps95ixcbbpip5kbvyzaad8r0-cairo-1.18.2/lib" "-L" "/nix/store/0713644mps95ixcbbpip5kbvyzaad8r0-cairo-1.18.2/lib" "-L" "/nix/store/7ly9p55j21sr0rl93lb6vpi1k1sbb3x3-glib-2.82.4/lib" "-L" "/nix/store/0713644mps95ixcbbpip5kbvyzaad8r0-cairo-1.18.2/lib" "-L" "/nix/store/7ly9p55j21sr0rl93lb6vpi1k1sbb3x3-glib-2.82.4/lib" "-L" "/nix/store/3fnl15840l1pjj7k6yypwv3hxdsp5wam-pango-1.56.1/lib" "-L" "/nix/store/zr9wda9pmmzrkv8hisck3vplcpds7k7b-harfbuzz-10.2.0/lib" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/build/libspa-0b9554779fb10caf/out" "-L" "/nix/store/nvj68rjl3f4900dba9gxynabzqg5m85n-pipewire-1.4.0/lib" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/build/libspa-sys-9bba8281a6542bea/out" "-L" "/nix/store/nvj68rjl3f4900dba9gxynabzqg5m85n-pipewire-1.4.0/lib" "-L" "/nix/store/464j7y43imzxc94955xgysxxcnadaah0-rustc-1.85.0/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/deps/niri-07cfe1b99baf688c" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-nodefaultlibs" "-Wl,--push-state,--no-as-needed" "-lEGL" "-lwayland-client" "-Wl,--pop-state" "/build/source/target/x86_64-unknown-linux-gnu/release/deps/niri_audit_data.o" "-Wl,--undefined=AUDITABLE_VERSION_INFO"
       >   = note: some arguments are omitted. use `--verbose` to show all linker arguments
       >   = note: /nix/store/dh0m67pv8snfcx1ic18jzp0x4cb10zhc-binutils-2.43.1/bin/ld: cannot find -lgbm: No such file or directory
       >           collect2: error: ld returned 1 exit status
       >           
       >
       > error: could not compile `niri` (bin "niri") due to 1 previous error
       For full logs, run 'nix log /nix/store/i3rpb32sga5sq7xc1awglv4wyyvhgb7x-niri-e251ca7.drv'.
error: 1 dependencies of derivation '/nix/store/hx1kb3s05yr0889wa1ri738mirxnpcpv-desktops.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3ia4wjphn1ih3xbwmlxzxhf2m4ksi7q7-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hiyz3sw55aks5zqyz4ipqfmdx7hziyni-niri-e251ca7_fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2p5b10rpizr4365zvhbjsxryg077s39w-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bbgvnnndya7vycf53gnx4ga6msxc96b4-nixos-system-wks-jb03-nix-25.05.20250318.b6eaf97.drv' failed to build
```

I'm not sure what's wrong but when searching for help, I noticed that there is now a package [libgbm](https://search.nixos.org/packages?channel=unstable&show=libgbm&from=0&size=50&sort=relevance&type=packages&query=libgbm) in nixpkgs-unstable. As mesa is included with comment "libgbm" my guess is, that this has to be included now instead?

I'll try to remove nixpkgs.follows for niri and see if that fixes it, but maybe this helps in case the next flake.lock update breaks it. I'll update when my compile finishes

--- a/flake.nix
+++ b/flake.nix
@@ -32,7 +32,7 @@
           libinput,
           seatd,
           libxkbcommon,
-          mesa,
+          libgbm,
           pango,
           pipewire,
           pkg-config,
@@ -92,7 +92,7 @@
               libinput
               seatd
               libxkbcommon
-              mesa # libgbm
+              libgbm
               pango
               wayland
             ]

